src/stream/src/common/log_store_impl/kv_log_store/reader.rs

```diff
@@ -43,6 +46,52 @@ use crate::common::log_store_impl::kv_log_store::serde::{
 };
 use crate::common::log_store_impl::kv_log_store::KvLogStoreMetrics;
 
+type RewindBackoffPolicy = impl Iterator<Item = Duration>;
+const REWIND_BASE_DELAY: Duration = Duration::from_secs(1);
+const REWIND_BACKOFF_FACTOR: u64 = 2;
+const REWIND_MAX_DELAY: Duration = Duration::from_secs(180);
+
+fn initial_rewind_backoff_policy() -> RewindBackoffPolicy {
+    tokio_retry::strategy::ExponentialBackoff::from_millis(REWIND_BASE_DELAY.as_millis() as _)
+        .factor(REWIND_BACKOFF_FACTOR)
+        .max_delay(REWIND_MAX_DELAY)
+        .map(tokio_retry::strategy::jitter)
+}
+
+struct RewindDelay {
+    last_rewind_truncate_offset: Option<TruncateOffset>,
+    backoff_policy: RewindBackoffPolicy,
+}
+
+impl RewindDelay {
+    fn new() -> Self {
+        Self {
+            last_rewind_truncate_offset: None,
+            backoff_policy: initial_rewind_backoff_policy(),
+        }
+    }
+
+    async fn rewind_delay(&mut self, truncate_offset: Option<TruncateOffset>) {
+        match (&self.last_rewind_truncate_offset, &truncate_offset) {
+            (Some(prev_rewind_truncate_offset), Some(truncate_offset)) => {
+                if truncate_offset > prev_rewind_truncate_offset {
+                    self.last_rewind_truncate_offset = Some(*truncate_offset);
+                    // Have new truncate progress before this round of rewind.
+                    // Reset rewind backoff
+                    self.backoff_policy = initial_rewind_backoff_policy();
```

Comment:
is it for another error that occurs when rewinding the prev one?

Reply:
No. The current logic is that, if the rewind is caused by the same error in the external sink, we will keep rewinding with exponential backoff delay. If the rewind is caused by a new error, we will reset to the initial delay.